### PR TITLE
Feature: added SEO meta tags in separate SEO component

### PR DESCRIPTION
**Pull request by lazarnikolov94**
Added SEO component with ordinary, Facebook and Twitter SEO meta tags. Could be improved for every template individually, but this works for now.

### CODE REVIEW

File: src/components/SEO.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  ogTitle?: string;
+  ogDescription?: string;
+  ogImage?: string;
+  ogUrl?: string;
+  twitterTitle?: string;
+  twitterDescription?: string;
+  twitterImage?: string;
+};
+
+export const SEO = (props: SEOProps) => {
```
Comment (hauptrolle) [inline src/components/SEO.tsx:27]: I would destructure the props here, and assign them a default value. So we can rid if all the `??`

File: src/components/SEO.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      <meta property="og:title" content={props.title ?? SEO_TITLE} />
+      <meta
+        property="og:description"
+        content={props.description ?? SEO_DESCRIPTION}
+      />
+      <meta property="og:image" content={props.ogImage ?? SEO_OG_IMAGE} />
+      <meta property="og:site_name" content={SEO_TITLE} />
+      <meta
+        property="og:url"
+        content={props.ogUrl ?? `https://chakra-templates.dev${router.asPath}`}
```
Comment (hauptrolle) [inline src/components/SEO.tsx:44]: Is the hardcoded URL here necessary, or does a relative one work aswell? If not please move the url to the constants

File: src/constants.ts [hunk 3]
```diff
@@ -20,3 +20,14 @@ export const SUGGEST_TEMPLATE_LINK =
...
+/**
+ * SEO constants
+ */
+export const SEO_TITLE = 'Chakra Templates';
+export const SEO_DESCRIPTION =
+  'A growing open-source collection of hand-crafted Chakra UI templates ready to drop into your React project.';
+export const SEO_KEYWORDS =
+  'chakra ui, ui templates, web templates, web design, react templates, react ui templates, chakra ui templates, chakra templates';
+export const SEO_OG_IMAGE = 'https://chakra-templates.dev/favicon.png';
+export const SEO_TWITTER_IMAGE = 'https://chakra-templates.dev/favicon.png';
```
Comment (hauptrolle) [inline src/constants.ts:33]: We should design some cool OG/Twitter images in the future :)

File: src/pages/[...slug].tsx [hunk 4]
```diff
@@ -23,12 +24,11 @@ const Templates: NextPage<PageProps> = ({
   templates,
   code,
 }) => {
+  const seoTitle = `${category.name}/${subCategory.name} - Chakra Templates`;
```
Comment (hauptrolle) [inline src/pages/[...slug].tsx:27]: You could use `SEO_TITLE` from the constants instead of the `Chakra Templates` here

**Review by hauptrolle — CHANGES_REQUESTED** [review]
Overall, cool stuff! 🥳 

I have just some small remarks 🙈

File: src/components/SEO.tsx (hunk 1 above)
Comment (lazarnikolov94) [reply src/components/SEO.tsx:27]: Yeah that sounds good

File: src/components/SEO.tsx (hunk 2 above)
Comment (lazarnikolov94) [reply src/components/SEO.tsx:44]: definitely

File: src/constants.ts (hunk 3 above)
Comment (lazarnikolov94) [reply src/constants.ts:33]: That's exactly what I was planning to do! 😁

File: src/pages/[...slug].tsx (hunk 4 above)
Comment (lazarnikolov94) [reply src/pages/[...slug].tsx:27]: True

**Review by hauptrolle — APPROVED** [review]

**Pull request merged**